modules/foreman_provisioning_template.py

```diff
@@ -0,0 +1,387 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_provisioning_template
+short_description: Manage Provisioning Template in Foreman
+description:
+    - "Manage Foreman Provisioning Template"
+    - "Uses https://github.com/SatelliteQE/nailgun"
+    - "Uses ansible_nailgun_cement in /module_utils"
+version_added: "2.4"
+author:
+- "Bernhard Hopfenmueller(@Fobhep)"
+- "Matthias Dellweg (@mdellweg)"
+requirements:
+    - nailgun >= 0.29.0
+options:
+    server_url:
+        description:
+        - URL of Foreman server
+        required: true
+    username:
+        description:
+        - Username on Foreman server
+        required: true
+        default: true
+    password:
+        description:
+        - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+        - Verify SSL of the Foreman server
+        required: false
+    audit_comment:
+        description:
+        - Content of the audit comment field
+        required: false
+    kind:
+        description:
+        - The provisioning template kind
+        required: false
+        choices:
+        -finish
+        -iPXE
+        -job_template
+        -POAP
+        -provision
+        -ptable
+        -PXELinux
+        -PXEGrub
+        -PXEGrub2
+        -script
+        -snippet
+        -user_data
+        -ZTP
+    template:
+        description:
+        - |
+            The content of the provisioning template, either this or file_name
+            is required as a source for the Provisioning Template "content".
+        required: false
+    file_name:
+        description:
+        - |
+            The path of a template file, that shall be imported.
+            Either this or template is required as a source for
+            the Provisioning Template "content".
+        required: false
+    locations:
+        description:
+        - The locations the template should be assigend to
+        required: false
+    locked:
+        description:
+        - Determines whether the template shall be locked
+        required: false
+        defaul: false
+        choices:
+        - true
+        - false
+    name:
+        description:
+        - |
+            The name a template should be assigned with in Foreman.
+            A name must be provided.
+            Possible sources are, ordererd by preference:
+            The "name" parameter, config header (inline or in a file), basename of a file.
+        required: false
+    organizations:
+        description:
+        - The organizations the template shall be assigned to
+        required: false
+    oses:
```

Comment:
when lower-cased, this is not immediately recognized as OSes, can we name this `operatingsystems`?